contracts/button/lib.rs

```diff
@@ -36,25 +36,13 @@ pub mod button_game {
     /// Event type
     type Event = <ButtonGame as ContractEventBase>::Type;
 
-    /// Event emitted when TheButton is created
-    #[ink(event)]
-    #[derive(Debug)]
-    pub struct ButtonCreated {
-        #[ink(topic)]
-        reward_token: AccountId,
-        #[ink(topic)]
-        ticket_token: AccountId,
-        start: BlockNumber,
-        deadline: BlockNumber,
-    }
-
     /// Event emitted when TheButton is pressed
     #[ink(event)]
     #[derive(Debug)]
     pub struct ButtonPressed {
         #[ink(topic)]
         by: AccountId,
-        when: BlockNumber,
+        round: Round,
```

Comment:
`Round` is less granular than `BlockNumber`, right? why is it better then?

Reply:
Block number is already available to the indexer, it's attached to every event. With a round we can e.g. create an endpoint that returns players scores in a given round.